### DISCUSSION TITLE
How to secure W11 when using WSL?

### DISCUSSION THREAD

**Discussion by hj-collab** [HotCakeX/Harden-Windows-Security#76] [Q&A]
Hi. How to secure W11 with WSL2? Any program from WSL2 can easily bypass WDAC and Windows firewall protection.

**Comment by HotCakeX** [comment]
Hi,
Compartmentalization is the key,

If you want to use everything on the same host/OS then I suggest applying the Harden Windows security script + [Strict Kernel-mode WDAC policy](https://github.com/HotCakeX/Harden-Windows-Security/wiki/WDAC-policy-for-BYOVD-Kernel-mode-only-protection)

That gives you very decent protection.

WSL executables are in the [Microsoft recommended block rules](https://learn.microsoft.com/en-us/windows/security/application-security/application-control/windows-defender-application-control/design/applications-that-can-bypass-wdac), so if you want to use them you need to use another environment that's not protected by User-mode WDAC policies. Use another physical device or Hyper-V VM with nested virtualization on the same device.
